src/Currencies/AggregateCurrencies.php

```diff
@@ -47,13 +48,10 @@ public function subunitFor(Currency $currency): int
     /** {@inheritDoc} */
     public function getIterator(): Traversable
     {
-        /** @psalm-var AppendIterator&Traversable<int|string, Currency> $iterator */
         $iterator = new AppendIterator();
 
         foreach ($this->currencies as $currencies) {
-            $currencyIterator = $currencies->getIterator();
-            /** @psalm-var AppendIterator&Traversable<int|string, Currency> $currencyIterator */
-            $iterator->append($currencyIterator);
+            $iterator->append(new IteratorIterator($currencies->getIterator()));
```

Comment:
@frederikbosch This was the only code change, but I think it's fine, as the test also passes.